src/canmatrix/compare.py

```diff
@@ -306,163 +326,164 @@ def compareFrame(f1, f2, ignore=None):
         if f1.comment is None:
             f1.add_comment("")
         if f1.comment != f2.comment:
-            result.addChild(
-                compareResult(
+            result.add_child(
+                CompareResult(
                     "changed", "FRAME", f1, [
                         "comment: " + f1.comment, "comment: " + f2.comment]))
 
     for s2 in f2.signals:
         s1 = f1.signal_by_name(s2.name)
         if not s1:
-            result.addChild(compareResult("added", "SIGNAL", s2))
+            result.add_child(CompareResult("added", "SIGNAL", s2))
 
-    if ignore is not None and "ATTRIBUTE" in ignore and ignore[
-            "ATTRIBUTE"] == "*":
+    if ignore is not None and "ATTRIBUTE" in ignore and ignore["ATTRIBUTE"] == "*":
         pass
     else:
-        result.addChild(compareAttributes(f1, f2, ignore))
+        result.add_child(compare_attributes(f1, f2, ignore))
 
     temp = [str(item) for item in f2.transmitters]
     for transmitter in f1.transmitters:
         if transmitter not in temp:
-            result.addChild(compareResult("removed", "Frame-Transmitter", f1))
+            result.add_child(CompareResult("removed", "Frame-Transmitter", f1))
 
     temp = [str(item) for item in f1.transmitters]
     for transmitter in f2.transmitters:
         if transmitter not in temp:
-            result.addChild(compareResult("added", "Frame-Transmitter",  f2))
+            result.add_child(CompareResult("added", "Frame-Transmitter", f2))
 
     for sg1 in f1.signalGroups:
         sg2 = f2.signal_group_by_name(sg1.name)
         if sg2 is None:
-            result.addChild(compareResult("removed", "Signalgroup", sg1))
+            result.add_child(CompareResult("removed", "Signalgroup", sg1))
         else:
-            result.addChild(compareSignalGroup(sg1, sg2))
+            result.add_child(compare_signal_group(sg1, sg2))
 
     for sg2 in f2.signalGroups:
         if f1.signal_group_by_name(sg2.name) is None:
-            result.addChild(compareResult("added", "Signalgroup", sg2))
+            result.add_child(CompareResult("added", "Signalgroup", sg2))
     return result
 
 
-def compareSignal(s1, s2, ignore=None):
-    result = compareResult("equal", "SIGNAL", s1)
+def compare_signal(s1, s2, ignore=None):
+    # type: (cm.Signal, cm.Signal, typing.Optional[typing.Mapping]) -> CompareResult
```

Comment:
It's always the same long `typing.Optional[typing.Mapping[str, typing.Union[str, bool]]]` and this function is rather "private". I think it's enough to check the mapping on the top level instead of copy pasting endless lines (which are than moreover reported as 'line too long'). I don't know.

Reply:
If type hints can't handle breaking onto multiple lines in comments (my first check failed) then I guess we just accept and ignore that factor.  But I would tend to be complete.

Reply:
My already committed proposal is a type alias, because the same Mapping is used multiple times within the file.